### DISCUSSION TITLE
QUploader filter image by width and/or height

### DISCUSSION THREAD

**Discussion by hooman-mirghasemi** [quasarframework/quasar#13339] [General - Components / Directives / etc]
Hi
How can i make a filter for check if image width and height lower than 500px dont let user to upload image?

I try to use filter and file have not width/ height so i try to make image object like this:
`
async function checkFileSize (files) {
 

   let images 
   files.map(async file => {
      let _URL = window.URL || window.webkitURL;
      let img = new Image();
      let objectUrl = _URL.createObjectURL(file);
      img.src = objectUrl;
      await img.decode();
      images.push(img)
   })

  return images.filter( img => {
    
     if (img.width < 500 || img.height < 500) {
       alert('so small')
       return false
     }
    return true
  })
}
`

but i got this error in console:

`
runtime-core.esm-bundler.js?f781:218 Uncaught TypeError: filteredFiles.includes is not a function
    at eval (use-file.js?7df7:157:1)
    at eval (use-file.js?7df7:9:1)
    at Array.forEach (<anonymous>)
    at filterFiles (use-file.js?7df7:8:1)
    at processFiles (use-file.js?7df7:156:1)
    at addFilesToQueue (uploader-core.js?cb75:265:1)
    at callWithErrorHandling (runtime-core.esm-bundler.js?f781:155:1)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js?f781:164:1)
    at HTMLInputElement.invoker (runtime-dom.esm-bundler.js?9cec:366:1)

`

and also i think the filter function only should filter files not images!

so how can i do it?

**Comment by hooman-mirghasemi** [comment]
It seems filter only accept normal function not aysinc func how can i get image width or hight when image object after load have these properties??

**Reply by pdanpdan** [reply]
I would let that filter alone and:

- listen for the `added` event
- in the added event make your check
- if it is not good the call `.removeFile(file)` on QUploader

**Comment by hooman-mirghasemi** [comment]
@rstoenescu 
why filter dosent accept async function?
working with files normally want some time ...
